### PR TITLE
fix(storage): Add storage area typing for storage keys

### PR DESCRIPTION
**Pull request by TheOnlyTails**
This introduces new types that enforce the `area:key` format of storage keys by the WXT storage APIs.

### CODE REVIEW

File: pnpm-lock.yaml
Comment (aklinker1) [inline pnpm-lock.yaml:1]: I don't think any of these changes are relevant, can you revert the changes to this file?

File: packages/wxt/src/__tests__/storage.test.ts
Comment (aklinker1) [inline packages/wxt/src/__tests__/storage.test.ts:1]: Could you add another test with a bad key to ensure types are working? Something like this should be good enough, can probably go outside the main describe for each storage area

```ts
it("should not accept keys without a valid storage area prefix", () => {
  // @ts-expect-error: Test passes if there is a type error here
  storage.getItem("test");
});
```

**Review by aklinker1 — CHANGES_REQUESTED** [review]
Thanks for the PR! Good change.

A couple of changes and I can get this merged!

File: packages/wxt/src/__tests__/storage.test.ts
Comment (TheOnlyTails) [reply packages/wxt/src/__tests__/storage.test.ts:1]: Done!

File: pnpm-lock.yaml
Comment (TheOnlyTails) [reply pnpm-lock.yaml:1]: done

**Comment by TheOnlyTails** [comment]
This is a bit of a nitpick, but do you know why the auto-generated docs show this in the definition of `StorageItemKey`?
image
https://deploy-preview-708--creative-fairy-df92c4.netlify.app/api/reference/wxt/storage/type-aliases/StorageItemKey.html#type-alias-storageitemkey

**Review by aklinker1 — APPROVED** [review]
> This is a bit of a nitpick, but do you know why the auto-generated docs show this in the definition of `StorageItemKey`?
> image

Haha nope, that's crazy. I probably need to upgrade the tool, but it's touchy and breaks horribly whenever I try.

**Comment by aklinker1** [comment]
Released in [`v0.18.5`](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.18.5)

**Pull request merged**